File: SEPT14/FLOOR14.py
def val(n):
	ret = n*(n+1)*(2*n+1)*(3*n*(n+1) - 1)
	return int(ret/30);

def main():
	t = int(raw_input())
	while(t > 0):
		N,M = map(int, raw_input().split())

		imax = N;
		imin = N/2;
		rem = N;
		cr = 2;
		ans = 0;
		summa = 0;

		while(rem>0):

			while(rem>0 and imax-imin>5):
				summa = (cr-1) * ( val(imax) - val(imin) );
				ans  = ans + summa;
				cr   += 1
				rem -= (imax-imin);
				imax = imin;
				imin = N/cr;
				if(imin==imax):
					imin -= 1;

			if(rem <= 0):
				break;
			print cr, imin, imax;
			if(rem <=1000000):
				for i in range(1, imax+1):
					ans = ans + pow(i, 4)*(N/i);
				break;
			for i in range(imin+1, imax+1):
				ans = ans + pow(i,4)*int(N/i);
			cr += 1;
			rem -= (imax-imin);
			imax = imin;
			imin = N/cr;
			if(imin == imax):
				imin -= 1;

		print ans%M;

		t -= 1;

if __name__=="__main__":
	main()

File: SEPT14/FLOOR14.cpp
/*
basic idea
there are n/2 nums till n which have floor(N/i) as 1
like this count group of nums which have same floor(N/i)

let x = N
number of nums with 1 as floor(N/i) = x - (N/2) => N-N/2   = N/2
x = N/2;
number of nums with 2 as floor(N/i) = x - (N/3) => N/2-N/3 = N/6
x = N/3
number of nums with 3 as floor(N/i) = x - (N/4) => N/3-N/4 = N/12
x = N/4
number of nums with 4 as floor(N/i) = x - (N/5) => N/4-N/5 = N/20
.
.
.
.
.
x = N/sqrt(N)
number of nums with ... as floor(N/i) = x - ...


// sum of fourth powers
a(n) = n*(1+n)*(1+2*n)*(-1+3*n+3*n^2)/30

100000000 100000

https://oeis.org/A064604

*/

#include <bits/stdc++.h>
using namespace std;

#define LL long long

LL compute(LL x, LL m){
	LL ret = (  );
}

(x*(1+x)*(1+2*x)*(-1+3*x+3*pow(x,2))/30)

int main()
{
	LL n;
	cin >> n >> m;
	LL imax = n;
	LL imin = n/2;
	LL cr = 2;
	while(imax - imin > 5){

	}

	return 0;
}